test/jdk/java/util/stream/GathererAPITest.java

```diff
@@ -0,0 +1,204 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Supplier;
+import java.util.stream.*;
+import java.util.stream.Gatherer;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.*;
+
+/**
+ * @test
+ * @summary Testing public API of Gatherer
+ * @enablePreview
+ * @run junit GathererAPITest
+ */
+
+public class GathererAPITest {
+    final static Supplier<Void> initializer = () -> (Void)null;
+    final static Gatherer.Integrator<Void, Integer, Integer> integrator = (v,e,d) -> d.push(e);
+    final static BinaryOperator<Void> combiner = (l,r) -> l;
+    final static BiConsumer<Void,Gatherer.Downstream<? super Integer>> finisher = (v,d) -> {};
+
+    final static Supplier<Void> nullInitializer = null;
+    final static Gatherer.Integrator<Void, Integer, Integer> nullIntegrator = null;
+    final static BinaryOperator<Void> nullCombiner = null;
+    final static BiConsumer<Void,Gatherer.Downstream<? super Integer>> nullFinisher = null;
+
+    private final static <T> Gatherer<T,?,T> passthrough() {
+        return Gatherer.of(
+                () -> (Void)null,
+                Gatherer.Integrator.<Void,T,T>ofGreedy((v,e,d) -> d.push(e)),
+                (l,r) -> l,
+                (v,d) -> {}
+        );
+    }
+
+    private final static void assertThrowsNPE(Supplier<? extends Object> supplier) {
+        try {
+            var discard = supplier.get();
+        } catch (NullPointerException npe) {
+            return;
+        }
+        fail("Expected NullPointerException but wasn't thrown!");
+    }
+
+    private final static <T> void assertThrowsUOE(Supplier<T> supplier) {
+        try {
+            var discard = supplier.get();
+        } catch (UnsupportedOperationException uoe) {
+            return;
+        }
+        fail("Expected NullPointerException but wasn't thrown!");
```

Comment:
If you are updating this test then you could use assertThrows(UnsupportedOperationException.class, supplier::get) here, same in assertThrowsNPE.  It's just a minor thing as I'm reading through some of the tests.